lib/ocpp/v201/charge_point.cpp

```diff
@@ -4058,21 +4061,30 @@ ChargePoint::set_variables(const std::vector<SetVariableData>& set_variable_data
 }
 
 GetCompositeScheduleResponse ChargePoint::get_composite_schedule(const GetCompositeScheduleRequest& request) {
-    return this->get_composite_schedule_internal(request);
+    std::set<ChargingProfilePurposeEnum> purposes_to_ignore = utils::get_purposes_to_ignore(
+        this->device_model->get_optional_value<std::string>(ControllerComponentVariables::IgnoredProfilePurposesOffline)
+            .value_or(""),
+        this->is_offline());
+    return this->get_composite_schedule_internal(request, purposes_to_ignore);
```

Comment:
is there an EVerest style guide covering the use of `this->`?
it is often needed in lambdas but here it shouldn't be needed.

Reply:
There is no style guide yet. Im coming from java (where `this` is required) and personally like it to explicitly mark the members.